delete_handler.go

```diff
@@ -152,48 +168,44 @@ func ParseDeleteArguments(args []string, fallBackFunc func()) (result DeleteComm
 }
 
 // TODO : unit tests
-func deleteBeforeTarget(walFolder StorageFolder, target *Backup, findFull bool, backups []BackupTime, dryRun bool) {
+func deleteBeforeTarget(walFolder StorageFolder, target *Backup, backupToScan []BackupTime, dryRun bool) {
 	backupFolder := target.BaseBackupFolder
-	dto, err := target.fetchSentinel()
+	allBackups, garbage, err := getBackupsAndGarbage(backupFolder)
 	if err != nil {
 		tracelog.ErrorLogger.FatalError(err)
 	}
-	if dto.isIncremental() {
-		if findFull {
-			target.Name = *dto.IncrementFullName
-		} else {
-			tracelog.ErrorLogger.Fatalf("%v is incremental and it's predecessors cannot be deleted. Consider FIND_FULL option.", target.Name)
-		}
+	if backupToScan == nil { // TODO : anti-pattern, needs refactoring
+		backupToScan = allBackups
 	}
-	var garbage []string
-	if backups == nil {
-		backups, garbage, err = getBackupsAndGarbage(backupFolder)
-		if err != nil {
-			tracelog.ErrorLogger.FatalError(err)
-		}
+	garbageToDelete := findGarbageToDelete(garbage, target)
+
+	if dryRun { // TODO : split this function by two: 'find objects to delete' and 'delete these objects'
+		tracelog.InfoLogger.Printf("Dry run finished.\n")
+		return
 	}
 
-	skipLine, walSkipFileName := ComputeDeletionSkipline(backups, target)
+	for _, garbageName := range garbageToDelete {
+		dropBackup(backupFolder, garbageName)
+	}
+	skipLine, walSkipFileName := ComputeDeletionSkipline(backupToScan, target)
+	if skipLine < len(backupToScan)-1 {
+		deleteWALBefore(walSkipFileName, walFolder)
```

Comment:
technically you can have only one DeleteWAL call, but yeah, it is safer this way